sklearn/tests/test_metaestimators.py

```diff
@@ -36,6 +37,9 @@ def __init__(self, name, construct, skip_methods=(),
                   skip_methods=['transform', 'inverse_transform', 'score']),
     DelegatorData('RFECV', RFECV,
                   skip_methods=['transform', 'inverse_transform', 'score']),
+    DelegatorData('BaggingClassifier', lambda est: BaggingClassifier(est),
+                  skip_methods=['transform', 'inverse_transform', 'score',
+                                'predict_proba', 'predict_log_proba', 'predict'])
```

Comment:
On 01/21/2015 01:45 AM, Gilles Louppe wrote:
>
> In sklearn/tests/test_metaestimators.py 
> <https://github.com/scikit-learn/scikit-learn/pull/4137#discussion_r23280888>:
>
> > @@ -36,6 +37,9 @@ def __init__(self, name, construct, skip_methods=(),
> >                    skip_methods=['transform', 'inverse_transform', 'score']),
> >      DelegatorData('RFECV', RFECV,
> >                    skip_methods=['transform', 'inverse_transform', 'score']),
> > +    DelegatorData('BaggingClassifier', lambda est: BaggingClassifier(est),
> > +                  skip_methods=['transform', 'inverse_transform', 'score',
> > +                                'predict_proba', 'predict_log_proba', 'predict'])
>
> I am not sure to understand what this PR is about, but yes, if that is 
> your question, |BaggingClassifier.predict| relies on 
> |BaggingClassifier.predict_proba| (which itself relies on 
> |predict_proba| of the base estimator, or to |predict| if the latter 
> does not support it).
>
This PR is about using the new @if_delegate_has_method decorator,
that makes ``hasattr(BaggingClassifier(), 'decision_function')`` work as 
expected.